docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,421 @@
+  return WithSubscription;
+}
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+```
+
+
+## Caveats
+
+Higher-order components come with a few caveats that aren't immediately obvious if you're new to React.
+
+### Don't Use HOCs Inside the render Method
+
+React's diffing algorithm (called reconciliation) uses component identity to determine whether it should update the existing subtree or throw it away and mount a new one. If the component returned from `render` is identical (`===`) to the component from the previous render, React recursively updates the subtree by diffing it with the new one. If they're not equal, the previous subtree is unmounted completely.
+
+Normally, you shouldn't need to think about this. But it matters for HOCs because it means you can't apply an HOC to a component within the render method of a component:
+
+```js
+render() {
+  // A new version of EnhancedComponent is created on every render
+  // EnhancedComponent1 !== EnhancedComponent2
+  const EnhancedComponent = enhance(MyComponent);
+  // That causes the entire subtree to unmount/remount each time!
+  return <EnhancedComponent />;
+}
+```
+
+The problem here isn't just about performance — remounting a component causes the state of that component and all of its children to be lost.
+
+Instead, apply HOCs outside the component definition so that the resulting component is created only once. Then, its identity will be consistent across renders. This is usually what you want, anyway.
+
+In those rare cases where you need to apply an HOC dynamically, you can also do it inside a component's lifecycle methods or its constructor.
+
+### Static Methods Must Be Copied Over
+
+Sometimes it's useful to define a static method on a React component. For example, Relay containers expose a static method `getFragment` to facilitate the composition of GraphQL fragments.
+
+When you apply an HOC to a component, though, the original component is wrapped with a container component. That means the new component does not have any of the static methods of the original component.
+
+```js
+// Define a static method
+WrappedComponent.staticMethod = function() {/*...*/}
+// Now apply an HOC
+const EnhancedComponent = enhance(WrappedComponent);
+
+// The enhanced component has no static method
+typeof EnhancedComponent.staticMethod === 'undefined' // true
+```
+
+To solve this, you could copy the methods onto the container before returning it
+
+```js
+function enhance(WrappedComponent) {
+  class Enhance extends React.Component {/*...*/}
+  // Must know exactly which method(s) to copy :(
+  Enhance.staticMethod = WrappedComponent.staticMethod;
+  return Enhance;
+}
+```
+
+but this requires you to know exactly which methods need to be copied. You can use [hoist-react-non-statics](https://github.com/mridgway/hoist-non-react-statics) to automatically copy all non-React static methods:
```

Comment:
no "but" at the beginning, capitalize This. Just for consistency with the phrasing around other inserted code blocks